pkg/build/graph/edges.go

```diff
@@ -22,22 +21,24 @@ const (
 
 // AddInputOutputEdges links the build config to other nodes for the images and source repositories it depends on.
 func AddInputOutputEdges(g osgraph.MutableUniqueGraph, node *buildgraph.BuildConfigNode) *buildgraph.BuildConfigNode {
-	output := node.BuildConfig.Parameters.Output
+	output := node.BuildConfig.Spec.Output
 	to := output.To
 	switch {
-	case to != nil && len(to.Name) > 0:
-		out := imagegraph.FindOrCreateSyntheticImageStreamTagNode(g, imagegraph.MakeImageStreamTagObjectMeta(defaultNamespace(to.Namespace, node.BuildConfig.Namespace), to.Name, output.Tag))
+	case to == nil:
+	case to.Kind == "DockerImage":
+		out := imagegraph.EnsureDockerRepositoryNode(g, to.Name, "")
 		g.AddEdge(node, out, BuildOutputEdgeKind)
-	case len(output.DockerImageReference) > 0:
-		out := imagegraph.EnsureDockerRepositoryNode(g, output.DockerImageReference, output.Tag)
+	case to.Kind == "ImageStreamTag":
+		bits := strings.Split(to.Name, ":")
```

Comment:
this method doesn't have the ability to return errors and i'm not refactoring the whole thing to do it, nor do i feel comfortable defaulting the missing tag...you pass invalid data, you get a panic.